cmd/alarms/main.go

```diff
@@ -0,0 +1,187 @@
+	}
+
+	var exitCode int
+	defer smqlog.ExitWithError(&exitCode)
+
+	tp, err := jaeger.NewProvider(ctx, svcName, cfg.JaegerURL, cfg.InstanceID, cfg.TraceRatio)
+	if err != nil {
+		logger.Error(fmt.Sprintf("failed to init Jaeger: %s", err))
+		exitCode = 1
+		return
+	}
+	defer func() {
+		if err := tp.Shutdown(ctx); err != nil {
+			logger.Error(fmt.Sprintf("error shutting down tracer provider: %v", err))
+		}
+	}()
+	tracer := tp.Tracer(svcName)
+
+	dbConfig := postgres.Config{Name: defDB}
+	if err := env.ParseWithOptions(&dbConfig, env.Options{Prefix: envPrefixDB}); err != nil {
+		logger.Error(err.Error())
+	}
+
+	db, err := postgres.Setup(dbConfig, *alarmsRepo.Migration())
+	if err != nil {
+		logger.Error(err.Error())
+		exitCode = 1
+		return
+	}
+	defer db.Close()
+
+	repo := alarmsRepo.NewAlarmsRepo(db)
+
+	authConfig := grpcclient.Config{}
+	if err := env.ParseWithOptions(&authConfig, env.Options{Prefix: envPrefixAuth}); err != nil {
+		logger.Error(fmt.Sprintf("failed to load %s auth configuration : %s", svcName, err))
+		exitCode = 1
+		return
+	}
+	authn, authnClient, err := authsvc.NewAuthentication(ctx, authConfig)
+	if err != nil {
+		logger.Error(err.Error())
+		exitCode = 1
+		return
+	}
+	defer authnClient.Close()
+	logger.Info("AuthN  successfully connected to auth gRPC server " + authnClient.Secure())
+
+	domsGrpcCfg := grpcclient.Config{}
+	if err := env.ParseWithOptions(&domsGrpcCfg, env.Options{Prefix: envPrefixDomains}); err != nil {
+		logger.Error(fmt.Sprintf("failed to load domains gRPC client configuration : %s", err))
+		exitCode = 1
+		return
+	}
+
+	domAuthz, _, domainsHandler, err := domainsAuthz.NewAuthorization(ctx, domsGrpcCfg)
+	if err != nil {
+		logger.Error(err.Error())
+		exitCode = 1
+		return
+	}
+	defer domainsHandler.Close()
+
+	authz, authzHandler, err := authsvcAuthz.NewAuthorization(ctx, authConfig, domAuthz)
+	if err != nil {
+		logger.Error("failed to create authz " + err.Error())
+		exitCode = 1
+		return
+	}
+	defer authzHandler.Close()
+
+	logger.Info("AuthZ successfully connected to auth gRPC server " + authzHandler.Secure())
+
+	idp := uuid.New()
+
+	svc := alarms.NewService(idp, repo)
+
+	svc = middleware.NewAuthorizationMiddleware(svc, authz)
+	svc = middleware.NewLoggingMiddleware(logger, svc)
+	counter, latency := prometheus.MakeMetrics("alarms", "api")
+	svc = middleware.NewMetricsMiddleware(counter, latency, svc)
+	svc = middleware.NewTracingMiddleware(tracer, svc)
+
+	httpServerConfig := server.Config{Port: defSvcHTTPPort}
+	if err := env.ParseWithOptions(&httpServerConfig, env.Options{Prefix: envPrefixHTTP}); err != nil {
+		logger.Error(fmt.Sprintf("failed to load %s HTTP server configuration : %s", svcName, err))
+		exitCode = 1
+		return
+	}
+	hs := httpserver.NewServer(ctx, cancel, svcName, httpServerConfig, httpAPI.MakeHandler(svc, logger, idp, cfg.InstanceID, authn), logger)
+
+	pubSub, err := brokers.NewPubSub(ctx, cfg.BrokerURL, logger)
```

Comment:
This PubSub defaults to broker with messaging configuration.  
i.e If broker is NATS, then stream with configuration  `Name: channels` and `Subjects: []string{"channels.>"},`   

We need something like `github.com/absmach/supermq/pkg/events` for `alarms`